lib/ecto/adapters/mysql/connection.ex

```diff
@@ -253,6 +254,22 @@ if Code.ensure_loaded?(Mariaex) do
       [" FROM ", from, " AS ", name | Enum.map(hints, &[?\s | &1])]
     end
 
+    defp cte(%{with: nil}, _), do: []
+    defp cte(%{with: %WithExpr{queries: []}}, _), do: []
+
+    defp cte(%{with: %WithExpr{recursive: recursive, queries: queries}} = query, sources) do
+      recursive_opt = recursive && "RECURSIVE " || ""
```

Comment:
I find this a bit clearer to read:

```suggestion
      recursive_opt = if recursive, do: "RECURSIVE ", else: ""
```

Thoughts?